pkg/querier/worker/scheduler_processor.go

```diff
@@ -357,24 +355,31 @@ func streamResponse(ctx context.Context, c client.PoolClient, queryID uint64, re
 		return fmt.Errorf("error sending initial response to frontend: %w", err)
 	}
 
+	// The response metadata has been sent successfully. After this point we can no longer
+	// return an error from this function as that would cause the response metadata to be sent
+	// again. This would be rejected by the frontend and the retry could never succeed.
+sendBody:
 	// Send body chunks.
 	for offset := 0; offset < len(response.Body); {
-		err = sc.Send(&frontendv2pb.QueryResultStreamRequest{
-			QueryID: queryID,
-			Data: &frontendv2pb.QueryResultStreamRequest_Body{Body: &frontendv2pb.QueryResultBody{
-				Chunk: response.Body[offset:min(offset+responseStreamingBodyChunkSizeBytes, len(response.Body))],
-			}},
-		})
-		if err != nil {
-			return fmt.Errorf("error sending response body chunk to frontend: %w", err)
+		select {
+		case <-reqCtx.Done():
+			break sendBody
+		default:
+			err = sc.Send(&frontendv2pb.QueryResultStreamRequest{
+				QueryID: queryID,
+				Data: &frontendv2pb.QueryResultStreamRequest_Body{Body: &frontendv2pb.QueryResultBody{
+					Chunk: response.Body[offset:min(offset+responseStreamingBodyChunkSizeBytes, len(response.Body))],
+				}},
+			})
+			if err != nil {
+				break sendBody
```

Comment:
let's somehow uncover this; otherwise, we won't know if queries are being aborted by the querier and we won't know why. I think a log line would suffice. 

If i could overengineer it I would return a special error type/kind from `streamResponse` which is interpreted in `runRequest` and prevents retries (and is logged along with its query ID as it is now). But I leave this up to you 😄

Reply:
I have added a log line for both of the `break`s to surface this condition being hit. I think it makes sense to understand possible failure modes first and then and add retry logic if needed.